File: admin/src/main/java/com/project/admin/domain/entity/TechBlogApproval.java
package com.project.admin.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class TechBlogApproval extends BaseEntity {

    @Id
    private Long id;

    @MapsId
    @JoinColumn(name = "tech_blog_id")
    @OneToOne
    private TechBlog techBlog;

    @Column(columnDefinition = "text")
    private String summary;

    public void update(String summary) {
        this.summary = summary;
    }

    public TechBlogApproval(TechBlog techBlog) {
        this.techBlog = techBlog;
    }
}

File: api/src/main/kotlin/com/project/api/entity/TechBlogApproval.kt
package com.project.api.entity

import jakarta.persistence.*

@Entity
@Table(name = "tech_blog_approval")
class TechBlogApproval {

    @Id
     val id: Long? = null

    @MapsId
    @JoinColumn(name = "tech_blog_id")
    @OneToOne(fetch = FetchType.LAZY)
     var techBlog: TechBlog? = null

    @Column(columnDefinition = "text")
     var summary: String? = null
}